### ISSUE TITLE
[SUGGESTION]: Additional filter for held items

### ISSUE THREAD

**Issue by MountTai1** [pokeclicker/pokeclicker#4191]
### Current Behavior

Currently the game filters held items based on the common filter options like stats, breeding, id etc.

![image](https://user-images.githubusercontent.com/131846168/234478889-895dfb9d-386c-4f17-bec0-3cae24bdcf71.png)

### Improved Behavior

I would like an additional filter like is done when applying vitamines to filter based on region.

![image](https://user-images.githubusercontent.com/131846168/234478029-dc92a887-2d0a-4aa1-8cd8-092b4282963b.png)

### Reason for change

The reason is that me (and I assume many others) only know about a few generations of pokemon so as soon as you enter into unfamiliar regions it is very difficult to figure out which pokemons are best to use your limited amount of held items on for the region you are in. 

### Detailed Description

By not being able to filter regions it makes it difficult for me to figure out which pokemon are from the region I am in and do not get debuffed attack.

As it is now I have to go through the trouble of first sorting by pokemon id, write down the names (or have another tab open on the internet with all pokemon from a region) and then filter on breeding and look for the right names. A region filter would save players a lot of trouble.

Basically I would just like the list to not be cluttered with pokemon from all regions but be able to select pokemon from 1 region.
